l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,286 @@
+            calls[cnt] = baseTokenRegistrationCall;
+            cnt++;
+        } else {
+            calls = new IGovernance.Call[](1);
+        }
+
+        IGovernance.Call memory registerChainCall = prepareRegisterHyperchainCall();
+        calls[cnt] = registerChainCall;
+        cnt++;
+
+        address l2SharedBridgeProxy = computeL2BridgeAddress();
+        IGovernance.Call memory initChainCall = prepareInitializeChainGovernanceCall(l2SharedBridgeProxy);
+
+        scheduleTransparentCalldata(calls, initChainCall);
+    }
+
+    function initializeConfig() internal {
+        // Grab config from output of l1 deployment
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script-config/prepare-registration-calldata.toml");
+        string memory toml = vm.readFile(path);
+
+        // Config file must be parsed key by key, otherwise values returned
+        // are parsed alfabetically and not by key.
+        // https://book.getfoundry.sh/cheatcodes/parse-toml
+        config.bridgehub = toml.readAddress("$.deployed_addresses.bridgehub_proxy_addr");
+        config.stateTransitionProxy = toml.readAddress("$.deployed_addresses.state_transition_proxy_addr");
+        config.l1SharedBridgeProxy = toml.readAddress("$.deployed_addresses.l1_shared_bridge_proxy_addr");
+        config.erc20BridgeProxy = toml.readAddress("$.deployed_addresses.erc20_bridge_proxy_addr");
+
+        config.chainId = toml.readUint("$.chain.chain_id");
+        config.eraChainId = toml.readUint("$.chain.era_chain_id");
+        config.chainAdmin = toml.readAddress("$.chain.admin");
+        config.diamondCutData = toml.readBytes("$.chain.diamond_cut_data");
+        config.bridgehubCreateNewChainSalt = toml.readUint("$.chain.bridgehub_create_new_chain_salt");
+        config.baseToken = toml.readAddress("$.chain.base_token_addr");
+        config.governance = toml.readAddress("$.chain.governance_addr");
+
+        bytecodes.l2SharedBridgeBytecode = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/contracts/bridge/L2SharedBridge.sol/L2SharedBridge.json"
+        );
+
+        bytecodes.l2SharedBridgeProxyBytecode = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol/TransparentUpgradeableProxy.json"
+        );
+        bytecodes.beaconProxy = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol/BeaconProxy.json"
+        );
+    }
+
+    function checkBaseTokenAddress() internal view {
+        if (config.baseToken == address(0)) {
+            revert("Token address is not set");
+        }
+
+        // Check if it's ethereum address
+        if (config.baseToken == ADDRESS_ONE) {
+            return;
+        }
+
+        if (config.baseToken.code.length == 0) {
```

Comment:
rather than 'isEOA' - we should be checking some 'basic' calls from ERC20 (like maybe 'name' or 'balance' etc).

Reply:
I tried doing that, but then the script crashes not allowing us to provide a readable revert reason
And try/catches don't really work in foundry scripts, so I kept it as is for the sake of better UX

Reply:
ok - let's mention it in the comment then.